<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Data Science for Biological, Medical and Health Research - 29&nbsp; Time To Event / Survival Data</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./cox1.html" rel="next">
<link href="./multinomial.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">


<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
    <div class="container-fluid d-flex justify-content-between">
      <h1 class="quarto-secondary-nav-title"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">Time To Event / Survival Data</span></h1>
      <button type="button" class="quarto-btn-toggle btn" aria-label="Show secondary navigation">
        <i class="bi bi-chevron-right"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Data Science for Biological, Medical and Health Research</a> 
    </div>
      </div>
      <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">Introduction</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./incomplete.html" class="sidebar-item-text sidebar-link">This is work in progress</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./setup.html" class="sidebar-item-text sidebar-link">R Setup</a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./nhanes.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Building the <code>nh432</code> example</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./nh432cb.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Codebook for <code>nh432</code></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./431review1.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">431 Review: Comparing Means</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./431review2.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">431 Review: Comparing Rates</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./431review3.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">431 Review: Fitting Linear Models</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./smart.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">BRFSS SMART Data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./singleimputation.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Dealing with Missingness: Single Imputation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./datasummaries.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Summarizing <code>smart_cle1</code></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./anovasmart.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Analysis of Variance with SMART</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./anovaexamples.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Two-Way ANOVA and Interaction</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ancova_emphysema.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">11</span>&nbsp; <span class="chapter-title">Analysis of Covariance</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ancova_smart.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">12</span>&nbsp; <span class="chapter-title">Analysis of Covariance with the SMART data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./nonlinearity.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">13</span>&nbsp; <span class="chapter-title">Adding Non-linear Terms to a Linear Regression</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./olsfitting.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">14</span>&nbsp; <span class="chapter-title">Using <code>ols</code> to fit linear models</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./prostate1.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">15</span>&nbsp; <span class="chapter-title">A Model for Prostate Cancer</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./prostate2.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">16</span>&nbsp; <span class="chapter-title">Validating our Prostate Cancer Model</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./multiple_imp.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">17</span>&nbsp; <span class="chapter-title">Multiple Imputation and Linear Regression</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./tableone.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">18</span>&nbsp; <span class="chapter-title">Building Table 1</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./logistic1.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">19</span>&nbsp; <span class="chapter-title">Logistic Regression: The Foundations</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./logistic2.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">20</span>&nbsp; <span class="chapter-title">Logistic Regression with <code>glm</code></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./logistic3.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">21</span>&nbsp; <span class="chapter-title">Logistic Regression with <code>lrm</code></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./effectsize.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">22</span>&nbsp; <span class="chapter-title">Estimating and Interpreting Effect Sizes</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./binaryregression.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">23</span>&nbsp; <span class="chapter-title">Colorectal Cancer Screening and Some Special Cases</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./count1.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">24</span>&nbsp; <span class="chapter-title">Modeling a Count Outcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./count2.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">25</span>&nbsp; <span class="chapter-title">Negative Binomial Models for Count Data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./count3.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">26</span>&nbsp; <span class="chapter-title">Zero-Inflated Models for Count Data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./ordinaloutcome.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">27</span>&nbsp; <span class="chapter-title">Modeling an Ordinal Categorical Outcome</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./multinomial.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Multinomial Logistic Regression</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./survival_data.html" class="sidebar-item-text sidebar-link active"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">Time To Event / Survival Data</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cox1.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">30</span>&nbsp; <span class="chapter-title">Cox Regression Models, Part 1</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./cox2.html" class="sidebar-item-text sidebar-link"><span class="chapter-number">31</span>&nbsp; <span class="chapter-title">Cox Regression Models, Part 2</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">References</a>
  </div>
</li>
    </ul>
    </div>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#r-setup-used-here" id="toc-r-setup-used-here" class="nav-link active" data-scroll-target="#r-setup-used-here"><span class="toc-section-number">29.1</span>  R Setup Used Here</a>
  <ul class="collapse">
  <li><a href="#data-load" id="toc-data-load" class="nav-link" data-scroll-target="#data-load"><span class="toc-section-number">29.1.1</span>  Data Load</a></li>
  </ul></li>
  <li><a href="#an-outline-of-key-topics-discussed-in-these-notes" id="toc-an-outline-of-key-topics-discussed-in-these-notes" class="nav-link" data-scroll-target="#an-outline-of-key-topics-discussed-in-these-notes"><span class="toc-section-number">29.2</span>  An Outline of Key Topics Discussed in these Notes</a></li>
  <li><a href="#foundations-of-survival-analysis" id="toc-foundations-of-survival-analysis" class="nav-link" data-scroll-target="#foundations-of-survival-analysis"><span class="toc-section-number">29.3</span>  Foundations of Survival Analysis</a>
  <ul class="collapse">
  <li><a href="#the-survival-function-st" id="toc-the-survival-function-st" class="nav-link" data-scroll-target="#the-survival-function-st"><span class="toc-section-number">29.3.1</span>  The Survival Function, <span class="math inline">\(S(t)\)</span></a></li>
  <li><a href="#kaplan-meier-estimator-of-the-survival-function" id="toc-kaplan-meier-estimator-of-the-survival-function" class="nav-link" data-scroll-target="#kaplan-meier-estimator-of-the-survival-function"><span class="toc-section-number">29.3.2</span>  Kaplan-Meier Estimator of the Survival Function</a></li>
  <li><a href="#creating-a-survival-object-in-r" id="toc-creating-a-survival-object-in-r" class="nav-link" data-scroll-target="#creating-a-survival-object-in-r"><span class="toc-section-number">29.3.3</span>  Creating a Survival Object in R</a></li>
  </ul></li>
  <li><a href="#a-first-example-recurrent-lobar-intracerebral-hemorrhage" id="toc-a-first-example-recurrent-lobar-intracerebral-hemorrhage" class="nav-link" data-scroll-target="#a-first-example-recurrent-lobar-intracerebral-hemorrhage"><span class="toc-section-number">29.4</span>  A First Example: Recurrent Lobar Intracerebral Hemorrhage</a></li>
  <li><a href="#building-a-survival-object" id="toc-building-a-survival-object" class="nav-link" data-scroll-target="#building-a-survival-object"><span class="toc-section-number">29.5</span>  Building a Survival Object</a></li>
  <li><a href="#kaplan-meier-estimate-of-the-survival-function" id="toc-kaplan-meier-estimate-of-the-survival-function" class="nav-link" data-scroll-target="#kaplan-meier-estimate-of-the-survival-function"><span class="toc-section-number">29.6</span>  Kaplan-Meier Estimate of the Survival Function</a>
  <ul class="collapse">
  <li><a href="#the-kaplan-meier-plot-using-base-r" id="toc-the-kaplan-meier-plot-using-base-r" class="nav-link" data-scroll-target="#the-kaplan-meier-plot-using-base-r"><span class="toc-section-number">29.6.1</span>  The Kaplan-Meier Plot, using Base R</a></li>
  <li><a href="#using-survminer-to-draw-survival-curves" id="toc-using-survminer-to-draw-survival-curves" class="nav-link" data-scroll-target="#using-survminer-to-draw-survival-curves"><span class="toc-section-number">29.6.2</span>  Using <code>survminer</code> to draw survival curves</a></li>
  <li><a href="#a-fancy-k-m-plot-with-a-number-at-risk-table" id="toc-a-fancy-k-m-plot-with-a-number-at-risk-table" class="nav-link" data-scroll-target="#a-fancy-k-m-plot-with-a-number-at-risk-table"><span class="toc-section-number">29.6.3</span>  A “Fancy” K-M Plot with a number at risk table</a></li>
  </ul></li>
  <li><a href="#comparing-survival-across-the-two-genotypes" id="toc-comparing-survival-across-the-two-genotypes" class="nav-link" data-scroll-target="#comparing-survival-across-the-two-genotypes"><span class="toc-section-number">29.7</span>  Comparing Survival Across the Two Genotypes</a>
  <ul class="collapse">
  <li><a href="#kaplan-meier-survival-function-estimates-by-genotype" id="toc-kaplan-meier-survival-function-estimates-by-genotype" class="nav-link" data-scroll-target="#kaplan-meier-survival-function-estimates-by-genotype"><span class="toc-section-number">29.7.1</span>  Kaplan-Meier Survival Function Estimates, by Genotype</a></li>
  </ul></li>
  <li><a href="#testing-the-difference-between-two-survival-curves" id="toc-testing-the-difference-between-two-survival-curves" class="nav-link" data-scroll-target="#testing-the-difference-between-two-survival-curves"><span class="toc-section-number">29.8</span>  Testing the difference between two survival curves</a>
  <ul class="collapse">
  <li><a href="#alternative-log-rank-tests" id="toc-alternative-log-rank-tests" class="nav-link" data-scroll-target="#alternative-log-rank-tests"><span class="toc-section-number">29.8.1</span>  Alternative log rank tests</a></li>
  </ul></li>
  <li><a href="#a-fancy-k-m-plot-with-a-number-at-risk-table-1" id="toc-a-fancy-k-m-plot-with-a-number-at-risk-table-1" class="nav-link" data-scroll-target="#a-fancy-k-m-plot-with-a-number-at-risk-table-1"><span class="toc-section-number">29.9</span>  A “Fancy” K-M Plot with a number at risk table</a>
  <ul class="collapse">
  <li><a href="#customizing-the-kaplan-meier-plot-presentation-further" id="toc-customizing-the-kaplan-meier-plot-presentation-further" class="nav-link" data-scroll-target="#customizing-the-kaplan-meier-plot-presentation-further"><span class="toc-section-number">29.9.1</span>  Customizing the Kaplan-Meier Plot Presentation Further</a></li>
  </ul></li>
  <li><a href="#the-hazard-function" id="toc-the-hazard-function" class="nav-link" data-scroll-target="#the-hazard-function"><span class="toc-section-number">29.10</span>  The Hazard Function</a>
  <ul class="collapse">
  <li><a href="#the-inverse-kaplan-meier-estimator-of-ht" id="toc-the-inverse-kaplan-meier-estimator-of-ht" class="nav-link" data-scroll-target="#the-inverse-kaplan-meier-estimator-of-ht"><span class="toc-section-number">29.10.1</span>  The Inverse Kaplan-Meier Estimator of <span class="math inline">\(H(t)\)</span></a></li>
  <li><a href="#cumulative-hazard-function-from-inverse-k-m" id="toc-cumulative-hazard-function-from-inverse-k-m" class="nav-link" data-scroll-target="#cumulative-hazard-function-from-inverse-k-m"><span class="toc-section-number">29.10.2</span>  Cumulative Hazard Function from Inverse K-M</a></li>
  <li><a href="#the-nelson-aalen-estimator-of-ht" id="toc-the-nelson-aalen-estimator-of-ht" class="nav-link" data-scroll-target="#the-nelson-aalen-estimator-of-ht"><span class="toc-section-number">29.10.3</span>  The Nelson-Aalen Estimator of <span class="math inline">\(H(t)\)</span></a></li>
  <li><a href="#convert-wide-data-to-long" id="toc-convert-wide-data-to-long" class="nav-link" data-scroll-target="#convert-wide-data-to-long"><span class="toc-section-number">29.10.4</span>  Convert Wide Data to Long</a></li>
  <li><a href="#plot-comparison-of-hazard-estimates" id="toc-plot-comparison-of-hazard-estimates" class="nav-link" data-scroll-target="#plot-comparison-of-hazard-estimates"><span class="toc-section-number">29.10.5</span>  Plot Comparison of Hazard Estimates</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title d-none d-lg-block"><span class="chapter-number">29</span>&nbsp; <span class="chapter-title">Time To Event / Survival Data</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>In many medical studies, the main outcome variable is the time to the occurrence of a particular event.</p>
<ul>
<li>In a randomized controlled trial of cancer, for instance, surgery, radiation, and chemotherapy might be compared with respect to time from randomization and the start of therapy until death.
<ul>
<li>In this case, the event of interest is the death of a patient, but in other situations it might be remission from a disease, relief from symptoms or the recurrence of a particular condition.</li>
<li>Such observations are generally referred to by the generic term <strong>survival data</strong> even when the endpoint or event being considered is not death but something else.</li>
</ul></li>
</ul>
<p>These notes on survival analysis are just an introduction to the key ideas of the field. The PQHS department offers an entire course on survival analysis (PQHS 435) and I recommend that to those of you interested in deeper learning about the approaches we’ll discuss, or in learning more about other approaches to survival analysis.</p>
<p>The OpenIntro Statistics extra material on <a href="https://www.openintro.org/book/surv_in_r/">Survival Analysis in R, written by David Diez</a> is a very useful guide to survival analysis in R, using the <code>survival</code> package and supplemented by the <code>KMsurv</code> and <code>OIsurv</code> packages. A PDF version of that material is available, along with a full set of the code used in that guide, along with instructions to obtain the packages.</p>
<section id="r-setup-used-here" class="level2" data-number="29.1">
<h2 data-number="29.1" class="anchored" data-anchor-id="r-setup-used-here"><span class="header-section-number">29.1</span> R Setup Used Here</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">comment =</span> <span class="cn">NA</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mosaic)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival) </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survminer)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse) </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="data-load" class="level3" data-number="29.1.1">
<h3 data-number="29.1.1" class="anchored" data-anchor-id="data-load"><span class="header-section-number">29.1.1</span> Data Load</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>hem <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"data/hem.csv"</span>, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="an-outline-of-key-topics-discussed-in-these-notes" class="level2" data-number="29.2">
<h2 data-number="29.2" class="anchored" data-anchor-id="an-outline-of-key-topics-discussed-in-these-notes"><span class="header-section-number">29.2</span> An Outline of Key Topics Discussed in these Notes</h2>
<p>In this chapter, we tackle the building blocks of survival analysis, and use R to work with survival objects.</p>
<ul>
<li>The Survival Function, <span class="math inline">\(S(t)\)</span>
<ul>
<li>The Kaplan-Meier Estimate/Plot</li>
<li>Comparing Survival Functions with log rank test</li>
</ul></li>
<li>The Hazard Function, <span class="math inline">\(H(t) = -log(S(t))\)</span></li>
<li>Using <code>survival</code> and related packages in R</li>
</ul>
<p>In the next chapter, we introduce the the Cox Proportional Hazards Regression Model, one of several available models for fitting regressions to time-to-event (survival) outcomes.</p>
</section>
<section id="foundations-of-survival-analysis" class="level2" data-number="29.3">
<h2 data-number="29.3" class="anchored" data-anchor-id="foundations-of-survival-analysis"><span class="header-section-number">29.3</span> Foundations of Survival Analysis</h2>
<p>Survival analysis is concerned with <em>prospective</em> studies, where we start with a cohort of subjects and follow them forwards in time to determine some clinical outcome. Follow-up continues until either some event of interest occurs, the study ends, or further observation becomes impossible.</p>
<p>The outcomes in a survival analysis consist of the subject’s <strong>fate</strong> and <strong>length of follow-up</strong> at the end of the study.</p>
<ul>
<li>For some patients, the outcome of interest may not occur during follow-up.</li>
<li>For such patients, whose follow-up time is <em>censored</em>, we know only that this event did not occur while the patient was being followed. We do not know whether or not it will occur at some later time.</li>
</ul>
<p>The primary problems with survival data are <em>non-normality</em> and <em>censoring</em>…</p>
<ol type="1">
<li><p>Survival data are quantitative, but not symmetrically distributed. They will often appear positively skewed, with a few people surviving a very long time compared with the majority; so assuming a normal distribution will not be reasonable.</p></li>
<li><p>At the completion of the study, some patients may not have reached the endpoint of interest (death, relapse, etc.). Consequently, the exact survival times are not known.</p>
<ul>
<li>All that is known is that the survival times are greater than the amount of time the individual has been in the study.</li>
<li>The survival times of these individuals are said to be <strong>censored</strong> (precisely, they are right-censored).</li>
</ul></li>
</ol>
<section id="the-survival-function-st" class="level3" data-number="29.3.1">
<h3 data-number="29.3.1" class="anchored" data-anchor-id="the-survival-function-st"><span class="header-section-number">29.3.1</span> The Survival Function, <span class="math inline">\(S(t)\)</span></h3>
<p>The <strong>survival function</strong>, <span class="math inline">\(S(t)\)</span> (sometimes called the survivor function) is the probability that the survival time, <span class="math inline">\(T\)</span>, is greater than or equal to a particular time, <span class="math inline">\(t\)</span>.</p>
<ul>
<li><span class="math inline">\(S(t)\)</span> = proportion of people surviving to time <span class="math inline">\(t\)</span> or beyond</li>
</ul>
<p>If there’s no censoring, the survival function is easy to estimate.</p>
<p><span class="math display">\[
\hat{S}(t) = \frac{\# \mbox{ of subjects with survival times } \geq t}{n}
\]</span></p>
<p>but this won’t work if there is censoring.</p>
</section>
<section id="kaplan-meier-estimator-of-the-survival-function" class="level3" data-number="29.3.2">
<h3 data-number="29.3.2" class="anchored" data-anchor-id="kaplan-meier-estimator-of-the-survival-function"><span class="header-section-number">29.3.2</span> Kaplan-Meier Estimator of the Survival Function</h3>
<p>The survival function <span class="math inline">\(S(t)\)</span> is the probability of surviving until at least time <span class="math inline">\(t\)</span>. It is essentially estimated by the number of patients alive at time <span class="math inline">\(t\)</span> divided by the total number of study subjects remaining at that time.</p>
<p>The Kaplan-Meier estimator first orders the (unique) survival times from smallest to largest, then estimates the survival function at each unique survival time.</p>
<ul>
<li>The survival function at the second death time, <span class="math inline">\(t_{(2)}\)</span> is equal to the estimated probability of not dying at time <span class="math inline">\(t_{(2)}\)</span> conditional on the individual being still at risk at time <span class="math inline">\(t_{(2)}\)</span>.</li>
</ul>
<p>In the presence of censoring, the survival function is estimated as follows.</p>
<ol type="1">
<li>Order the survival times from smallest to largest, where t_{(j)} is the <span class="math inline">\(j\)</span>th largest unique survival time, so we have…</li>
</ol>
<p><span class="math display">\[
t_{(1)} \leq t_{(2)} \leq t_{(3)} \leq ... t_{(n)}
\]</span></p>
<ol start="2" type="1">
<li>The Kaplan-Meier estimate of the survival function is</li>
</ol>
<p><span class="math display">\[
\hat{S}(t) = \prod_{j: t_{(j)} \leq t} (1 - \frac{d_j}{r_j})
\]</span></p>
<p>where <span class="math inline">\(r_j\)</span> is the number of people at risk just before <span class="math inline">\(t_{(j)}\)</span>, including those censored at time <span class="math inline">\(t_{(j)}\)</span>, and <span class="math inline">\(d_j\)</span> is the number of people who experience the event at time <span class="math inline">\(t_{(j)}\)</span>.</p>
<p>When we want to compare survival functions (or their Kaplan-Meier estimates, at least) we’ll use a <strong>log rank test</strong> or one of several extensions of that test.</p>
</section>
<section id="creating-a-survival-object-in-r" class="level3" data-number="29.3.3">
<h3 data-number="29.3.3" class="anchored" data-anchor-id="creating-a-survival-object-in-r"><span class="header-section-number">29.3.3</span> Creating a Survival Object in R</h3>
<p>To do survival analysis in R, we’re going to start with three main functions, all in the <code>survival</code> package:</p>
<ul>
<li><code>Surv</code> creates a survival object</li>
<li><code>survfit</code> builds a Kaplan-Meier test, and the results may be plotted, as we’ve seen.</li>
<li><code>survdiff</code> builds a log rank test, that will let us compare two survival functions, as well as running several alternatives.</li>
</ul>
<p>Plus, we’ll build out some estimates of the hazard function.</p>
<p>The <code>Surv</code> function, part of the <code>survival</code> package in R, will create a <strong>survival object</strong> from two arguments:</p>
<ol type="1">
<li><code>time</code> = follow-up time</li>
<li><code>event</code> = a status indicator, where
<ul>
<li>event = 1 or TRUE means the event was observed (for instance, the patient died)</li>
<li>event = 0 or FALSE means the follow-up time was censored</li>
</ul></li>
</ol>
</section>
</section>
<section id="a-first-example-recurrent-lobar-intracerebral-hemorrhage" class="level2" data-number="29.4">
<h2 data-number="29.4" class="anchored" data-anchor-id="a-first-example-recurrent-lobar-intracerebral-hemorrhage"><span class="header-section-number">29.4</span> A First Example: Recurrent Lobar Intracerebral Hemorrhage</h2>
<p>O’Donnell et al.&nbsp;(2000) studied the effect of the apolipoprotein E gene on the risk of recurrent lobar intracerebral hemorrhage in 70 patients who survived such a hemorrhage<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>. Patients in the study are classified by:</p>
<ul>
<li><code>time</code> = follow-up time, in months</li>
<li><code>recur</code> = indicator of whether or not they had a recurrent event (1 = subject had a recurrence, 0 subject did not have a recurrence), and</li>
<li><code>genotype</code> = the subject’s apolipoprotein E genotype (0 = Homozygous <span class="math inline">\(\epsilon3/\epsilon3\)</span> and 1 = At least one <span class="math inline">\(\epsilon2\)</span> or <span class="math inline">\(\epsilon4\)</span> allele)</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>hem <span class="sc">|&gt;</span> <span class="fu">head</span>(<span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4 × 4
     id genotype  time recur
  &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
1     1        0 0.230     1
2     2        0 1.05      0
3     3        1 1.38      0
4     4        1 1.41      1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(hem<span class="sc">$</span>recur)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 0  1 
52 18 </code></pre>
</div>
</div>
<p>We have 70 patients at the start, and observe 18 events (rest are censored.)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">favstats</span>(time <span class="sc">~</span> recur, <span class="at">data =</span> hem)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  recur       min        Q1   median       Q3      max     mean       sd  n
1     0 1.0513350 14.414783 23.01437 38.64477 53.88091 25.51129 14.68242 52
2     1 0.2299795  3.367556 10.92403 24.84600 42.87474 14.98517 13.88893 18
  missing
1       0
2       0</code></pre>
</div>
</div>
<p>The median survival time looks like 23 weeks in the patients who do not exhibit a recurrence, but only 11 weeks in those who do.</p>
</section>
<section id="building-a-survival-object" class="level2" data-number="29.5">
<h2 data-number="29.5" class="anchored" data-anchor-id="building-a-survival-object"><span class="header-section-number">29.5</span> Building a Survival Object</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>hemsurv <span class="ot">&lt;-</span> <span class="fu">Surv</span>(<span class="at">time =</span> hem<span class="sc">$</span>time, <span class="at">event =</span> hem<span class="sc">$</span>recur)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hemsurv, <span class="dv">4</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.2299795  1.0513350+ 1.3798770+ 1.4127311 </code></pre>
</div>
</div>
<p>This object both displays the survival time for each subject, and indicates whether or not the subject’s follow-up was <em>censored</em> before a recurrent event occurred. Survival times with a <code>+</code> sign indicate censoring.</p>
<ul>
<li>Subject 1 lived for 0.23 months hemorrhage-free and then had a recurrence.</li>
<li>Subject 2 lived for 1.05 months hemorrhage-free, at which point they were censored (perhaps because the study ended, or perhaps because the subject was no longer available for follow-up)</li>
</ul>
<p>Remember that 18 of the subjects experienced a recurrent hemorrhage, and the other 52 are therefore censored.</p>
</section>
<section id="kaplan-meier-estimate-of-the-survival-function" class="level2" data-number="29.6">
<h2 data-number="29.6" class="anchored" data-anchor-id="kaplan-meier-estimate-of-the-survival-function"><span class="header-section-number">29.6</span> Kaplan-Meier Estimate of the Survival Function</h2>
<p>To build a Kaplan-Meier estimate of the survival function (to account properly for censoring), we take the survival object we have created, and use the <code>survfit</code> function from the <code>survival</code> package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>hemfit1 <span class="ot">&lt;-</span> <span class="fu">survfit</span>(hemsurv <span class="sc">~</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We can look at the <code>hemfit1</code> object directly, although the K-M estimate is usually plotted.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(hemfit1, <span class="at">print.rmean=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Call: survfit(formula = hemsurv ~ 1)

      n events rmean* se(rmean) median 0.95LCL 0.95UCL
[1,] 70     18   40.4      2.63     NA    42.9      NA
    * restricted mean with upper limit =  53.9 </code></pre>
</div>
</div>
<p>We see that 18 events occurred out of a total of 70 subjects. The median survival time is listed as <code>NA</code> (missing) which implies it cannot be estimated by this simple model.</p>
<ul>
<li>This is because only 18 of our 70 subjects have a known recurrence-free survival time (the rest are censored), so we don’t actually know what the median survival time will be across our 70 subjects. Apparently, R can produce a lower bound on a 95% confidence interval for the median survival time, but not the actual point estimate.</li>
</ul>
<p>We also observe a <strong>restricted mean survival time</strong> estimate. The restricted mean uses as its upper limit the largest observed or censored survival time, which here is a censored value: 53.9 months. So it is the mean survival time, assuming all censored subjects lived hemorrhage-free for 53.9 months.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(hemfit1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Call: survfit(formula = hemsurv ~ 1)

  time n.risk n.event survival std.err lower 95% CI upper 95% CI
  0.23     70       1    0.986  0.0142        0.958        1.000
  1.41     67       1    0.971  0.0202        0.932        1.000
  1.58     65       1    0.956  0.0248        0.909        1.000
  3.06     63       1    0.941  0.0287        0.886        0.999
  3.32     62       1    0.926  0.0320        0.865        0.991
  3.52     61       1    0.911  0.0349        0.845        0.982
  3.55     60       1    0.895  0.0375        0.825        0.972
  4.76     57       1    0.880  0.0400        0.805        0.962
  9.53     54       1    0.863  0.0424        0.784        0.951
 12.32     50       1    0.846  0.0449        0.762        0.939
 15.57     46       1    0.828  0.0476        0.740        0.926
 19.15     38       1    0.806  0.0511        0.712        0.912
 24.77     32       1    0.781  0.0553        0.679        0.897
 24.87     31       1    0.756  0.0590        0.648        0.881
 28.09     26       1    0.726  0.0635        0.612        0.862
 33.61     22       1    0.693  0.0687        0.571        0.842
 37.52     17       1    0.653  0.0758        0.520        0.819
 42.87      8       1    0.571  0.1011        0.404        0.808</code></pre>
</div>
</div>
<p>This written summary provides us with lots of detail on the Kaplan-Meier estimate. In particular, the first two lines of this summary can be read to indicate the following.</p>
<ul>
<li>Up to time 0.23 months, no patients had a recurrence. Then, an event occurred, and the estimated survival (i.e.&nbsp;non-recurrence) probability is reduced from 1 to 0.986.</li>
<li>By time 1.41 months, when the next event occurred, only 67 patients remained at risk. This is because one of them had a recurrent hemorrhage already (at 0.23 months) and two others had been right-censored. The estimated hemorrhage-free survival probability estimate starting at time 1.41 months is now 0.971.</li>
</ul>
<p>A Kaplan-Meier plot graphically represents this summary.</p>
<section id="the-kaplan-meier-plot-using-base-r" class="level3" data-number="29.6.1">
<h3 data-number="29.6.1" class="anchored" data-anchor-id="the-kaplan-meier-plot-using-base-r"><span class="header-section-number">29.6.1</span> The Kaplan-Meier Plot, using Base R</h3>
<p>Now, let’s plot the Kaplan-Meier estimate, so we can see what is going on.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(hemfit1, <span class="at">ylab=</span><span class="st">"Pr(Hemorrhage-Free Survival)"</span>,</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"Months of Follow-Up"</span>,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">main=</span><span class="st">"Kaplan-Meier Plot for hem Data"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="survival_data_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>The solid line indicates estimated hemorrhage-free survival probability. The dotted lines identify pointwise confidence intervals (default 95%).</p>
<ul>
<li>For example, we see that the estimated probability of hemorrhage-free survival to 20 months is estimated to be about 0.8</li>
<li>The estimated probability of hemorrhage-free survival to 50 months is estimated to be about 0.6</li>
</ul>
<p>The steps down indicate events (recurrences.) The estimated probability of survival to 0 months starts at 1, and drops down at each time point where an event (or more than one event) is observed.</p>
</section>
<section id="using-survminer-to-draw-survival-curves" class="level3" data-number="29.6.2">
<h3 data-number="29.6.2" class="anchored" data-anchor-id="using-survminer-to-draw-survival-curves"><span class="header-section-number">29.6.2</span> Using <code>survminer</code> to draw survival curves</h3>
<p>Another approach to plotting the Kaplan-Meier estimate comes from <code>ggsurvplot</code>, from the <strong>survminer</strong> package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(hemfit1, <span class="at">data =</span> hem)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="survival_data_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Again, the solid line indicates estimated hemorrhage-free survival probability. The crosses indicate censoring. The steps down indicate events (recurrences,) and the shading indicates (default 95%) pointwise confidence intervals. By <strong>pointwise</strong> confidence intervals, I mean that these bounds apply only to individual points in the time scale.</p>
<p>For more on an alternative approach, using <em>simultaneous confidence bands</em>, visit the OpenIntro Statistics <a href="https://www.openintro.org/stat/surv.php">Survival Analysis in R materials, written by David Diez</a>, which are also posted on <a href="https://github.com/THOMASELOVE/432-2018/blob/master/texts/Diez_OpenIntro_survival_analysis_in_R.pdf">our web site</a>.</p>
</section>
<section id="a-fancy-k-m-plot-with-a-number-at-risk-table" class="level3" data-number="29.6.3">
<h3 data-number="29.6.3" class="anchored" data-anchor-id="a-fancy-k-m-plot-with-a-number-at-risk-table"><span class="header-section-number">29.6.3</span> A “Fancy” K-M Plot with a number at risk table</h3>
<p>We can do a lot more with these plots. Following the suggestions at https://github.com/kassambara/survminer/ we can create the following…</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(hemfit1, <span class="at">data =</span> hem,</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>   <span class="at">conf.int =</span> <span class="cn">TRUE</span>, <span class="co"># Add confidence interval</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">risk.table =</span> <span class="cn">TRUE</span>, <span class="co"># Add risk table</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">xlab =</span> <span class="st">"Time in months"</span>, <span class="co"># adjust X axis label</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">break.time.by =</span> <span class="dv">12</span> <span class="co"># add tick every 12 months</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>   )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="survival_data_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>This sort of plot is really designed to work best when we compare multiple groups in terms of their survival. So let’s do that.</p>
</section>
</section>
<section id="comparing-survival-across-the-two-genotypes" class="level2" data-number="29.7">
<h2 data-number="29.7" class="anchored" data-anchor-id="comparing-survival-across-the-two-genotypes"><span class="header-section-number">29.7</span> Comparing Survival Across the Two Genotypes</h2>
<p>Now, suppose we want to compare the hemorrhage-free survival functions for subjects classified by their apoliprotein E genotype. Working with the same survival object <code>hemsurv</code> we now run the <code>survfit</code> function to compare across the two <code>genotype</code> groups.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>hemfit2 <span class="ot">&lt;-</span> <span class="fu">survfit</span>(hemsurv <span class="sc">~</span> hem<span class="sc">$</span>genotype)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(hemfit2, <span class="at">print.rmean=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Call: survfit(formula = hemsurv ~ hem$genotype)

                n events rmean* se(rmean) median 0.95LCL 0.95UCL
hem$genotype=0 32      4   47.8      2.87     NA      NA      NA
hem$genotype=1 38     14   33.9      3.77   37.5    24.9      NA
    * restricted mean with upper limit =  53.9 </code></pre>
</div>
</div>
<ul>
<li>In <code>genotype = 0</code> (the subjects who are Homozygous <span class="math inline">\(\epsilon3/\epsilon3\)</span>,) we had 32 subjects, and observed 4 recurrent hemorrhages. Our estimated restricted mean survival time in those subjects is 44.8 months and we cannot estimate a median survival time because only a small fraction of our subjects were not censored.</li>
<li>In <code>genotype = 1</code> (subjects who have at least one <span class="math inline">\(\epsilon2\)</span> or <span class="math inline">\(\epsilon4\)</span> allele,) we had 38 subjects and observed 14 recurrences. The estimated restricted mean survival time is 32.7 months in these subjects, and we can (it seems) estimate a median survival time in this group of 37.5 months. Note that we don’t actually need to observe the event in half of the subjects to estimate a median survival time.</li>
</ul>
<section id="kaplan-meier-survival-function-estimates-by-genotype" class="level3" data-number="29.7.1">
<h3 data-number="29.7.1" class="anchored" data-anchor-id="kaplan-meier-survival-function-estimates-by-genotype"><span class="header-section-number">29.7.1</span> Kaplan-Meier Survival Function Estimates, by Genotype</h3>
<p>I find I have to crank the figure height in Quarto up to at least 6 to get the risk table to show up nicely in this setting.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(hemfit2, <span class="at">data =</span> hem,</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">conf.int =</span> <span class="cn">TRUE</span>,</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlab =</span> <span class="st">"Time in months"</span>,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">break.time.by =</span> <span class="dv">12</span>,</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">legend.labs =</span> <span class="fu">c</span>(<span class="st">"Homozygous"</span>, <span class="st">"Heterozygous"</span>),</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">risk.table =</span> <span class="cn">TRUE</span>,</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">risk.table.height =</span> <span class="fl">0.25</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>           )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="survival_data_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>It appears that patients who were homozygous for the <span class="math inline">\(\epsilon3\)</span> allele of this gene (i.e.&nbsp;<code>genotype</code> = 0 in the hemorrhage data) had a much better prognosis than others (<code>genotype</code> = 1.)</p>
</section>
</section>
<section id="testing-the-difference-between-two-survival-curves" class="level2" data-number="29.8">
<h2 data-number="29.8" class="anchored" data-anchor-id="testing-the-difference-between-two-survival-curves"><span class="header-section-number">29.8</span> Testing the difference between two survival curves</h2>
<p>To obtain a significance test comparing these two survival curves, we turn to a <em>log rank</em> test, which tests the null hypothesis <span class="math inline">\(H_0: S_1(t) = S_2(t)\)</span> for all <span class="math inline">\(t\)</span> where the two exposures have survival functions <span class="math inline">\(S_1(t)\)</span> and <span class="math inline">\(S_2(t)\)</span>. We use the <code>survdiff</code> function to explore this test, which uses a <span class="math inline">\(\chi^2\)</span> statistic to do the testing.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">survdiff</span>(hemsurv <span class="sc">~</span> hem<span class="sc">$</span>genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survdiff(formula = hemsurv ~ hem$genotype)

                N Observed Expected (O-E)^2/E (O-E)^2/V
hem$genotype=0 32        4     9.28      3.00      6.28
hem$genotype=1 38       14     8.72      3.19      6.28

 Chisq= 6.3  on 1 degrees of freedom, p= 0.01 </code></pre>
</div>
</div>
<p>Based on the log rank test, we conclude that there is a statistically significant difference (<span class="math inline">\(p\)</span> = .0122) between the hemorrhage-free survival curves for the two genotypes, as shown in the Kaplan-Meier plot.</p>
<ul>
<li>The log rank test generalizes to permit survival comparisons across more than two groups, with the test statistic having an asymptotic chi-squared distribution with one degree of freedom less than the number of patient groups being compared.</li>
</ul>
<section id="alternative-log-rank-tests" class="level3" data-number="29.8.1">
<h3 data-number="29.8.1" class="anchored" data-anchor-id="alternative-log-rank-tests"><span class="header-section-number">29.8.1</span> Alternative log rank tests</h3>
<p>An alternative approach to testing is the <em>Peto and Peto modification of the Gehan-Wilcoxon test</em>, which results from adding <code>rho=1</code> to the <code>survdiff</code> function (<code>rho=0</code>, the default, yields the log rank test.)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">survdiff</span>(hemsurv <span class="sc">~</span> hem<span class="sc">$</span>genotype, <span class="at">rho=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Call:
survdiff(formula = hemsurv ~ hem$genotype, rho = 1)

                N Observed Expected (O-E)^2/E (O-E)^2/V
hem$genotype=0 32     3.63     7.87      2.29      5.46
hem$genotype=1 38    11.79     7.54      2.39      5.46

 Chisq= 5.5  on 1 degrees of freedom, p= 0.02 </code></pre>
</div>
</div>
<p>As compared to the log rank test, this Peto-Peto modification (and others using <code>rho &gt; 0</code>) give greater weight to the left hand (earlier) side of the survival curves.</p>
<ul>
<li>To obtain chi-square tests that give greater weight to the right hand (later) side of the survival curves than the log rank test, use a <code>rho</code> value which is less than 0.</li>
</ul>
</section>
</section>
<section id="a-fancy-k-m-plot-with-a-number-at-risk-table-1" class="level2" data-number="29.9">
<h2 data-number="29.9" class="anchored" data-anchor-id="a-fancy-k-m-plot-with-a-number-at-risk-table-1"><span class="header-section-number">29.9</span> A “Fancy” K-M Plot with a number at risk table</h2>
<p>We can add the log rank test result to our “fancy” K-M plot. Visit https://github.com/kassambara/survminer/ for more options.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(hemfit2, <span class="at">data =</span> hem, <span class="at">size =</span> <span class="dv">1</span>,</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>   <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">"purple"</span>, <span class="st">"darkgoldenrod"</span>), <span class="co"># custom colors</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>   <span class="at">conf.int =</span> <span class="cn">TRUE</span>, <span class="co"># Add confidence interval</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>   <span class="at">pval =</span> <span class="cn">TRUE</span>, <span class="co"># Add p-value</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>   <span class="at">risk.table =</span> <span class="cn">TRUE</span>, <span class="co"># Add risk table</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>   <span class="at">risk.table.height =</span> <span class="fl">0.25</span>, <span class="co"># change if you have &gt;2 groups</span></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>   <span class="at">risk.table.y.text.col =</span> T, <span class="co"># show colors in table listing</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>   <span class="at">xlab =</span> <span class="st">"Time in months"</span>, <span class="co"># adjust X axis label</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>   <span class="at">break.time.by =</span> <span class="dv">12</span>, <span class="co"># break X axis in time intervals</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>   <span class="at">legend.labs =</span> <span class="fu">c</span>(<span class="st">"Homozygous"</span>, <span class="st">"Heterozygous"</span>), <span class="co"># labels</span></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>   <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>() <span class="co"># Change ggplot2 theme</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>   )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="survival_data_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<section id="customizing-the-kaplan-meier-plot-presentation-further" class="level3" data-number="29.9.1">
<h3 data-number="29.9.1" class="anchored" data-anchor-id="customizing-the-kaplan-meier-plot-presentation-further"><span class="header-section-number">29.9.1</span> Customizing the Kaplan-Meier Plot Presentation Further</h3>
<p>We can even add a plot of the number of censored subjects at each time point, as well as a median survival pointer (which, of course, we’ve seen that we can’t estimate in one of the groups), and customize the style of the confidence intervals. Again, see https://github.com/kassambara/survminer/ for even more customized results.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsurvplot</span>(hemfit2,              </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">data =</span> hem, </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">palette =</span> <span class="fu">c</span>(<span class="st">"#E7B800"</span>, <span class="st">"#2E9FDF"</span>),</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>           <span class="at">risk.table =</span> <span class="cn">TRUE</span>,      </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">pval =</span> <span class="cn">TRUE</span>,           </span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">conf.int =</span> <span class="cn">TRUE</span>,       </span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">xlab =</span> <span class="st">"Time in months"</span>, </span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>           <span class="at">break.time.by =</span> <span class="dv">12</span>,     </span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>           <span class="at">ggtheme =</span> <span class="fu">theme_light</span>(),</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>           <span class="at">risk.table.y.text.col =</span> T,</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>           <span class="at">risk.table.height =</span> <span class="fl">0.25</span>, </span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">risk.table.y.text =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>           <span class="at">ncensor.plot =</span> <span class="cn">TRUE</span>,</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>           <span class="at">ncensor.plot.height =</span> <span class="fl">0.25</span>,</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>           <span class="at">conf.int.style =</span> <span class="st">"step"</span>,</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>           <span class="at">surv.median.line =</span> <span class="st">"hv"</span>,</span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>           <span class="at">legend.labs =</span> <span class="fu">c</span>(<span class="st">"Homozygous"</span>, <span class="st">"Heterozygous"</span>)</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>        )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="survival_data_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
</section>
<section id="the-hazard-function" class="level2" data-number="29.10">
<h2 data-number="29.10" class="anchored" data-anchor-id="the-hazard-function"><span class="header-section-number">29.10</span> The Hazard Function</h2>
<p>To build regression models for time-to-event data, we will need to introduce the <strong>hazard function</strong>. Consider a subject in the hemorrhage study who has a hemorrhage-free survival time of 9 months.</p>
<ul>
<li>For this subject to have had a recurrent hemorrhage at 9 months, they had to be hemorrhage-free for the first 8 months.</li>
<li>The subject’s hazard at 9 months is the failure rate “per month” conditional on the subject being hemorrhage-free for the first 8 months.</li>
</ul>
<p>If <span class="math inline">\(S(t)\)</span> is the survival function, and time <span class="math inline">\(t\)</span> is taken to be continuous, then <span class="math inline">\(S(t) = e^{H(t)}\)</span> defines the hazard function <span class="math inline">\(H(t)\)</span>.</p>
<ul>
<li>Note that <span class="math inline">\(H(t)\)</span> = <span class="math inline">\(-ln(S(t))\)</span>.</li>
<li>The function <span class="math inline">\(H(t)\)</span> is an important analytic tool.
<ul>
<li>It’s used to describe the concept of the risk of “failure” in an interval after time <span class="math inline">\(t\)</span>, conditioned on the subject having survived to time <span class="math inline">\(t\)</span>.</li>
<li>It’s often called the <em>cumulative hazard function</em>, to emphasize the fact that its value is the “sum” of the hazard up to time <span class="math inline">\(t\)</span>.</li>
</ul></li>
</ul>
<p>There are several different methods to estimate <span class="math inline">\(H(t)\)</span>, but we’ll focus on two…</p>
<ol type="1">
<li>The inverse Kaplan-Meier estimator</li>
<li>The Nelson-Aalen estimator</li>
</ol>
<section id="the-inverse-kaplan-meier-estimator-of-ht" class="level3" data-number="29.10.1">
<h3 data-number="29.10.1" class="anchored" data-anchor-id="the-inverse-kaplan-meier-estimator-of-ht"><span class="header-section-number">29.10.1</span> The Inverse Kaplan-Meier Estimator of <span class="math inline">\(H(t)\)</span></h3>
<p>Our first estimator of the hazard function, <span class="math inline">\(H(t)\)</span> will be the inverse Kaplan-Meier estimate, which I’ll place in an R object called <code>H_est1</code>.</p>
<ul>
<li>To start, we will take the negative of the log of the Kaplan-Meier survival estimate. That takes care of the first <em>t</em>-1 levels of the eventual estimate.</li>
<li>To complete the process, we will repeat the final one of those time-specific estimates at the end.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>H_est1 <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log</span>(hemfit1<span class="sc">$</span>surv)</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>H_est1 <span class="ot">&lt;-</span> <span class="fu">c</span>(H_est1, <span class="fu">tail</span>(H_est1, <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Here are the first five, and last five values of the hazard function estimate.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(H_est1,<span class="dv">5</span>) <span class="co"># first 5 values</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.01438874 0.01438874 0.01438874 0.02942661 0.02942661</code></pre>
</div>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(H_est1, <span class="dv">5</span>) <span class="co"># last 5</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.5602049 0.5602049 0.5602049 0.5602049 0.5602049</code></pre>
</div>
</div>
<p>We can create a little tibble containing the times and hazard estimates, like this:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>haz_hem <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">time =</span> <span class="fu">c</span>(hemfit1<span class="sc">$</span>time, <span class="fu">tail</span>(hemfit1<span class="sc">$</span>time, <span class="dv">1</span>)),</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">inverse_KM =</span> H_est1</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="cumulative-hazard-function-from-inverse-k-m" class="level3" data-number="29.10.2">
<h3 data-number="29.10.2" class="anchored" data-anchor-id="cumulative-hazard-function-from-inverse-k-m"><span class="header-section-number">29.10.2</span> Cumulative Hazard Function from Inverse K-M</h3>
<p>Since we’ve built the data set of times and hazard values, we can use the <code>geom_step</code> function in <code>ggplot2</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(haz_hem, <span class="fu">aes</span>(<span class="at">x =</span> time, <span class="at">y =</span> inverse_KM)) <span class="sc">+</span> </span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_step</span>() <span class="sc">+</span> </span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>, <span class="dv">24</span>, <span class="dv">36</span>, <span class="dv">48</span>)) <span class="sc">+</span></span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Months of Follow-Up"</span>, </span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Cumulative Hazard"</span>,</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Cumulative Hazard Function via Inverse K-M"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="survival_data_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>
<section id="the-nelson-aalen-estimator-of-ht" class="level3" data-number="29.10.3">
<h3 data-number="29.10.3" class="anchored" data-anchor-id="the-nelson-aalen-estimator-of-ht"><span class="header-section-number">29.10.3</span> The Nelson-Aalen Estimator of <span class="math inline">\(H(t)\)</span></h3>
<p>An alternative estimate of the cumulative hazard is called the Nelson-Aalen estimate, captured here in <code>H_est2</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>h_st <span class="ot">&lt;-</span> hemfit1<span class="sc">$</span>n.event <span class="sc">/</span> hemfit1<span class="sc">$</span>n.risk</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>H_est2 <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(h_st)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>H_est2 <span class="ot">&lt;-</span> <span class="fu">c</span>(H_est2, <span class="fu">tail</span>(H_est2, <span class="dv">1</span>))</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>haz_hem<span class="sc">$</span>Nelson_Aalen <span class="ot">&lt;-</span> H_est2</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(haz_hem)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
   time inverse_KM Nelson_Aalen
  &lt;dbl&gt;      &lt;dbl&gt;        &lt;dbl&gt;
1 0.230     0.0144       0.0143
2 1.05      0.0144       0.0143
3 1.38      0.0144       0.0143
4 1.41      0.0294       0.0292
5 1.51      0.0294       0.0292
6 1.58      0.0449       0.0446</code></pre>
</div>
</div>
</section>
<section id="convert-wide-data-to-long" class="level3" data-number="29.10.4">
<h3 data-number="29.10.4" class="anchored" data-anchor-id="convert-wide-data-to-long"><span class="header-section-number">29.10.4</span> Convert Wide Data to Long</h3>
<p>In order to easily plot the two hazard function estimates in the same graph, we’ll want to convert these data from wide format to long format, with the <code>pivot_longer</code> function.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>haz_hem_comp <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(<span class="at">data =</span> haz_hem, <span class="at">cols =</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>,</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">names_to =</span> <span class="st">"method"</span>, <span class="at">values_to =</span> <span class="st">"hazardest"</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(haz_hem_comp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 3
   time method       hazardest
  &lt;dbl&gt; &lt;chr&gt;            &lt;dbl&gt;
1 0.230 inverse_KM      0.0144
2 0.230 Nelson_Aalen    0.0143
3 1.05  inverse_KM      0.0144
4 1.05  Nelson_Aalen    0.0143
5 1.38  inverse_KM      0.0144
6 1.38  Nelson_Aalen    0.0143</code></pre>
</div>
</div>
</section>
<section id="plot-comparison-of-hazard-estimates" class="level3" data-number="29.10.5">
<h3 data-number="29.10.5" class="anchored" data-anchor-id="plot-comparison-of-hazard-estimates"><span class="header-section-number">29.10.5</span> Plot Comparison of Hazard Estimates</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(haz_hem_comp, <span class="fu">aes</span>(<span class="at">x =</span> time, <span class="at">y =</span> hazardest, </span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">col =</span> method)) <span class="sc">+</span> </span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_step</span>(<span class="at">linewidth =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>, <span class="dv">24</span>, <span class="dv">36</span>, <span class="dv">48</span>)) <span class="sc">+</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Months of Follow-Up"</span>, </span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Cumulative Hazard"</span>,</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Cumulative Hazard Function"</span>) <span class="sc">+</span> </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="survival_data_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>We can see that the two cumulative hazard function estimates are nearly identical in this case. We could instead compare the two functions in faceted plots, if that would be helpful.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(haz_hem_comp, <span class="fu">aes</span>(<span class="at">x =</span> time, <span class="at">y =</span> hazardest)) <span class="sc">+</span> </span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_step</span>() <span class="sc">+</span> </span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">12</span>, <span class="dv">24</span>, <span class="dv">36</span>, <span class="dv">48</span>)) <span class="sc">+</span></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Months of Follow-Up"</span>, </span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">y =</span> <span class="st">"Cumulative Hazard"</span>,</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">title =</span> <span class="st">"Cumulative Hazard Function"</span>) <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">facet_grid</span>(method <span class="sc">~</span> .) <span class="sc">+</span> <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="survival_data_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>Next, we will consider the issue of modeling a survival outcome using Cox proportional hazards regression.</p>


</section>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p>These data come from Dupont WD (2002) Statistical Modeling for Biomedical Researchers. New York: Cambridge U. Press, section 6.4.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./multinomial.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">28</span>&nbsp; <span class="chapter-title">Multinomial Logistic Regression</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./cox1.html" class="pagination-link">
        <span class="nav-page-text"><span class="chapter-number">30</span>&nbsp; <span class="chapter-title">Cox Regression Models, Part 1</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->



</body></html>